doc/scheduling.md

```diff
@@ -0,0 +1,180 @@
+main `SRV_GROUP` option is mandatory.
+
+If whole `SRV_GROUP` group is offline, request
+will be scheduled to `BACKUP_SRV_GROUP`. Backup server groups are very handy for
+[A/B testing](https://en.wikipedia.org/wiki/A/B_testing) purposes: in that case
+`SRV_GROUP` can represent unstable/test service, and `BACKUP_SRV_GROUP` -- stable
+production service; if test service is under the maintenance users will be
+forwarded to stable service.
+
+`FIELD` -- an HTTP request field, following are supported:
+* **uri** Only a part of URI is looked at that contains the path and the query
+string if any. (e.g. `/abs/path.html?query&key=val#fragment`);
+* **host** The host part from URI in HTTP request line, or the value of `Host`
+header. Host part in URI takes priority over the `Host` header value;
+* **hdr_host** The value of `Host` header;
+* **hdr_conn**  The value of `Connection` header;
+* **hdr_raw** The contents of any other HTTP header field as specified by
+`ARG`. `ARG` must include contents of an HTTP header starting with the header
+field name. The `suffix` `OP` is not supported for this `FIELD`. Processing
+of `hdr_raw` may be slow because it requires walking over all headers of an
+HTTP request.
+
+`OP` -- a string comparison operator:
+* **eq** `FIELD` is fully equal to the string specified in `ARG`;
+* **prefix** `FIELD` starts with the string specified in `ARG`;
+* **suffix** `FIELD` ends with the string specified in `ARG`;
+
+`ARG` -- an argument for the operator `OP`.
+
+Example configuration:
+```
+srv_group static { ... }
+srv_group foo_app { ... }
+srv_group foo_beta { ... }
+srv_group bar_app { ... }
+
+sched_http_rules {
+	match static   uri       prefix  "/static" ;
+	match static   uri       suffix  ".php";
+	match static   host      prefix  "static.";
+	match static   host      suffix  "tempesta-tech.com";
+	match foo_beta host      eq      "beta-foo.example.com" backup=foo_app;
+	match foo_app  host      eq      "foo.example.com";
+	match bar_app  hdr_conn  eq      "keep-alive";
+	match bar_app  hdr_host  prefix  "bar.";
+	match bar_app  hdr_host  suffix  "natsys-lab.com";
+	match bar_app  hdr_host  eq      "bar.natsys-lab.com";
+	match bar_app  hdr_raw   prefix  "X-Custom-Bar-Hdr: ";
+}
+```
+
+**HTTP Scheduler** matches request to rules in _top-down order_. If requests
+fits some rule, rule is applied and further matching is skipped. User can provide
+default match rule that matches any request. If no default rule provided and
+[default server group]() is defined, default match rule to group `default` will
+be added implicitly to the end of the list. The syntax is as follows:
+```
+match <SRV_GROUP> * * * ;
+```
+
+
+## Scheduling Requests inside Server Groups
+
+**Server Schedulers** are bound to exact server groups, see
+[Setting scheduler]() section in [Server Group]() description.
+
+User can override default server scheduler for all groups using `sched`
+configuration option:
+```
+sched <SCHED_NAME>;
+```
```

Comment:
It's unclear where the option should be specified. It'd be useful to provide a configuration example where some scheduler is specified.